test/qa/test_meshnet.py

```diff
@@ -464,3 +455,64 @@ def test_invite_accept_non_existant_special_character():
         sh.nordvpn.meshnet.invite.accept("\u2222@test.com")
 
     assert "No invitation from '\u2222@test.com' was found." in str(ex.value)
+
+
+@pytest.mark.parametrize("meshnet_allias", meshnet.MESHNET_ALIAS)
+def test_set_meshnet_off_on(meshnet_allias):
+
+    assert "Meshnet is set to 'disabled' successfully." in sh.nordvpn.set(meshnet_allias, "off")
+    assert not settings.is_meshnet_on()
+
+    assert "Meshnet is set to 'enabled' successfully." in sh.nordvpn.set(meshnet_allias, "on")
+    assert settings.is_meshnet_on()
+
+
+@pytest.mark.parametrize("meshnet_allias", meshnet.MESHNET_ALIAS)
+def test_set_meshnet_on_repeated(meshnet_allias):
+
+    with pytest.raises(sh.ErrorReturnCode_1) as ex:
+            sh.nordvpn.set(meshnet_allias, "on")
+
+    assert "Meshnet is already enabled." in str(ex.value)
+
+
+@pytest.mark.parametrize("meshnet_allias", meshnet.MESHNET_ALIAS)
+def test_set_meshnet_off_repeated(meshnet_allias):
+
+    sh.nordvpn.set(meshnet_allias, "off")
+
+    with pytest.raises(sh.ErrorReturnCode_1) as ex:
+            sh.nordvpn.set(meshnet_allias, "off")
+
+    assert "Meshnet is already disabled." in str(ex.value)
+
+    sh.nordvpn.set.meshnet("on")
+
+
+@pytest.mark.parametrize("meshnet_allias", meshnet.MESHNET_ALIAS)
+def test_set_meshnet_on_logged_out(meshnet_allias):
```

Comment:
[minor] wouldn't it be better to add a `when`?  test_set_meshnet_on_`when`_logged_out ? same for the next one :smile:

Reply:
Yeah, probably would make a bit more sense :D